docs/source/nlp/nemo_megatron/retro/retro_model.rst

```diff
@@ -0,0 +1,501 @@
+        model.fp32_residual_connection=True \
+        +model.activations_checkpoint_granularity=full \
+        model.activations_checkpoint_method=block \
+        model.activations_checkpoint_num_layers=1
+
+During the training, launch Tensorboard to monitor training like so:
+
+.. code-block:: bash
+
+    tensorboard --logdir /result/retro_model --bind_all
+
+After the training, the model nemo file can be found at the result checkpoint directory.
+
+Option 2: Train the NeMo RETRO model *with* mu-Transfer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`mu-Transfer <https://openreview.net/pdf?id=Bx6qKuBM2AD>`_ paper proposed a method to zero-shot transfer hyperparameter to train a larger model.
+This can be done in 3 steps in NeMo RETRO implementation. 
+
+
+Step 1. find optimal hyper parameter for a small base model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the pre-training code in Option 1, either manually or automatically ind a set of optimal hyperparameter for a small base RETRO 
+model. This is can be done cheaply ans fast due to the small model size.
+
+Step 2. calculate the shape file that can be used to run mu-Transfer 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The shape file determines which hyperparameters will be scaled up, allowing the model to adjust the learning rate, weight scaling factor, etc.
+
+Here is an example shape file calculation script:
+
+.. code-block:: bash
+    python examples/nlp/language_modeling/megatron_retro_cal_shape.py
+    trainer.devices=8 \
+    trainer.num_nodes=1 \
+    trainer.accelerator=gpu \
+    trainer.accumulate_grad_batches=1 \
+    trainer.max_steps=500000 \
+    trainer.precision=bf16 \
+    trainer.val_check_interval=2000 \
+    trainer.limit_val_batches=200 \
+    trainer.gradient_clip_val=1.0 \
+    exp_manager.exp_dir=/result/retro_model \
+    base_model.enc_num_layers=2 \
+    delta_model.enc_num_layers=2 \
+    base_model.dec_num_layers=12 \
+    delta_model.dec_num_layers=12 \
+    base_model.dec_num_layers=32 \
+    delta_model.dec_num_layers=32 \
+    base_model.tensor_model_parallel_size=8 \
+    delta_model.tensor_model_parallel_size=8 \
+    base_model.dec_cross_attention=[8,11,14,17,20,23,26,29,31] \
+    delta_model.dec_cross_attention=[8,11,14,17,20,23,26,29,31] \
+    base_model.enc_cross_attention=[0] \
+    delta_model.enc_cross_attention=[0] \
+    base_model.hidden_size=768 \
+    base_model.ffn_hidden_size=3072 \
+    delta_model.hidden_size=96 \
+    delta_model.ffn_hidden_size=384 \
+    base_model.num_attention_heads=16 \
+    delta_model.num_attention_heads=16 \
+    base_model.add_position_embedding=False \
+    delta_model.add_position_embedding=False \
+    base_model.normalization=rmsnorm \
+    delta_model.normalization=rmsnorm \
+    base_model.megatron_amp_O2=False \
+    delta_model.megatron_amp_O2=False \
+    model.shape_file=tp8_32depth_o1_rel_shape_info.yaml 
```

Comment:
Same here, consider putting this into a config. I think it will be nicer to read in the docs.

Reply:
I remove most of the arguments and only leave the necessary ones.